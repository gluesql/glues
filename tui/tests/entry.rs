#[macro_use]
mod tester;
use tester::Tester;

use {color_eyre::Result, glues_tui::input::KeyCode};

#[tokio::test]
async fn opens_instant_on_enter() -> Result<()> {
    let mut t = Tester::new().await?;

    // initial draw (home)
    t.draw()?;

    // move down then up, then Enter to open Instant via selection
    t.press('j').await;
    t.press('k').await;
    t.key(KeyCode::Enter).await;

    // draw and snapshot notebook view (e.g., sample note visible)
    t.draw()?;
    snap!(t, "instant");

    Ok(())
}

#[tokio::test]
async fn quits_on_q() -> Result<()> {
    let mut t = Tester::new().await?;
    t.draw()?;

    let quit = t.press('q').await;
    assert!(quit);

    Ok(())
}

#[tokio::test]
async fn help_overlay_toggles() -> Result<()> {
    let mut t = Tester::new().await?;
    t.draw()?;

    // open help via shortcut
    t.press('h').await;
    t.draw()?;
    snap!(t, "help_open");

    // any key closes help
    t.press('x').await;
    t.draw()?;
    snap!(t, "help_closed");

    Ok(())
}

#[cfg(not(target_arch = "wasm32"))]
#[tokio::test]
async fn open_local_prompt() -> Result<()> {
    let mut t = Tester::new().await?;

    t.draw()?;
    // open Local storage prompt via hotkey 'l'
    t.press('l').await;
    t.draw()?;
    snap!(t, "local_prompt");

    Ok(())
}

#[tokio::test]
async fn proxy_prompt_open() -> Result<()> {
    let mut t = Tester::new().await?;
    t.draw()?;

    // open Proxy prompt via hotkey 'p'
    t.press('p').await;
    t.draw()?;
    snap!(t, "proxy_prompt_open");

    Ok(())
}
