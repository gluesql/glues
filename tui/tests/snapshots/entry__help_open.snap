---
source: tui/tests/entry.rs
assertion_line: 45
expression: text
snapshot_kind: text
---
 Glues - TUI note-taking app offering complete data control and flexible storage options               [?] Show keymap 
                                                                                                                        
┌─────────────────────────────────────────────────────────Help─────────────────────────────────────────────────────────┐
│                                                                                                                      │
│  Glues offers various storage options to suit your needs:                                                            │
│                                                                                                                      │
│  Instant                                                                                                             │
│  Data is stored in memory and only persists while the app is running.                                                │
│  This option is useful for testing or temporary notes as it is entirely volatile.                                    │
│                                                                                                                      │
│  Local                                                                                                               │
│  Notes are stored locally as separate files.                                                                         │
│  This is the default option for users who prefer a simple, file-based approach without any remote synchronization.   │
│                                                                                                                      │
│  redb                                                                                                                │
│  Persist everything inside a single redb database file.                                                              │
│  Provide a path and Glues will create or reuse the file for portable storage.                                        │
│                                                                                                                      │
│  Git                                                                                                                 │
│  Git storage requires three inputs: `path`, `remote`, and `branch`.                                                  │
│  The `path` should point to an existing local Git repository, similar to the file storage path.                      │
│  For example, you can clone a GitHub repository and use that path.                                                   │
│  The `remote` and `branch` specify the target remote repository and branch for synchronization.                      │
│  When you modify notes or directories, Glues will automatically sync changes with the specified remote repository.   │
│                                                                                                                      │
│  MongoDB                                                                                                             │
│  MongoDB storage allows you to store your notes in a MongoDB database, providing a scalable and centralized          │
│  solution for managing your notes.                                                                                   │
│  You need to provide the MongoDB connection string and the database name.                                            │
│  Glues will handle storing and retrieving notes from the specified database.                                         │
│                                                                                                                      │
│  Proxy                                                                                                               │
│  Connects Glues to a remote backend exposed over HTTP.                                                               │
│  Run `glues server memory` (or file/redb/git/mongo) to start a local proxy.                                          │
│                                                                                                                      │
│                                                                                                                      │
│                                                Press any key to close                                                │
│                                                                                                                      │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
